Automattic/gridicons
# Refactoring: simplified design/build process, SVG as master

![screen shot 2017-06-09 at 18 16 56](https://user-images.githubusercontent.com/4389/26986554-e486926e-4d3f-11e7-8880-f103d7e01861.png)

This PR is a proposals discussed with @drw158, @jasmussen, and @mattmiklic to simplify the process to add and update Gridicons.

Changes:

1. The master files are SVGs inside `/sources/svg/`
2. **No more AI files** — except for the template
3. **No more Illustrator scripts**
4. Since I was there, I also reviewed the comments in the `Gruntfile.js` to make it more readable.
5. Should also address the question raised in #206: instead of removing `svg/` we remove the AI files, the end result is more tight source SVGs.

Note that it seems a huge PR but most of the 180+ files are just files moved around.

### Benefits

* Faster turnaround of PRs as icons can be more easily merged
* Easier for anyone to add new icons
* Less binary files to handle
* Complete automation is possible

The only slight "con" is that SVGs might lose some editing capabilities the IA have, but since we started Gridicons in 2014 this issue never appeared in practice, and if anything we have used SVG as design sources.

### Historical Note

Initially we wanted to generate everything from Illustrator using Artboards, but we hit Illustrator limit to 100 Artboards (yes, really) and we reverted to single stack in #23. 
We also tried to use Sketch to preserve the ability to see all the icons side by side, but we had issue with that too, see #19 (mostly due to Sketch being not too evolved in SVG export at the time, today might be different).

Android also moved to use SVGs directly (with Android Studio) and iOS still requires a manual conversion to PDF, none of which again requires an Illustrator file. Due to these updated processes, the need for the special `gridicons-32.ai` file is reduced as tweaks should be done on a per-icon basis (see #14), and even there, we can just make `32` versions of the SVG files.

Thus, as long as the template is present to make sure the margins are followed (was added by Dave in #24), our real "master" copy of the icons are the SVG. 


### To test

1. Delete all temp folders: `node_modules`, `build` (even better: a clean git repository)
2. Readme: follow the "Propose a New Icon" steps (without creating a PR)
3. Readme: follow the "Add a Proposed Icon to Gridicons (Admins Only)" steps (without merging any PR)
4. Verify that all the necessary files are re-built as before (`svg-min`, `build`, `php`, `svg-sprite`, `docs`)
5. Try to remove the added icon (from both `sources/svg` and `svg-min`) and check in the diff that everything is restored to the initial state.
